app/controllers/admin/approvals_controller.rb

```diff
@@ -0,0 +1,8 @@
+class Admin::ApprovalsController < ApplicationController
+  def create
+    application_pet = ApplicationPet.find_by(application_id: params[:id], pet_id: params[:pet_id])
+    application_pet.approve
+
+    redirect_to "/admin/applications/#{application_pet.application.id}"
```

Comment:
I appreciated your defense of this design and why you chose to have a separate controller and make it a POST. It is more RESTfully conventional to direct that button to "/applications/:app_id/pets/:pet_id" and route that to `application_pets#update` BUT your defense makes a lot of sense and there is definitely a tradeoff between RESTfulness and user-friendly pathing. I agree that adding `/approve` to the path can make it more clear what the operation will do. This is a great example of when RESTfulness can be broken if you have a good defense.